source/Dig/base.util.scriptexpression_ng.bmx

```diff
@@ -0,0 +1,1693 @@
+	Return New SToken( TK_BOOLEAN, Long(params.GetToken(1).CompareWith(params.GetToken(2)) <= 0), first.linenum, first.linepos )
+End Function
+
+Function SEFN_Concat:SToken(params:STokenGroup Var, context:SScriptExpressionContext var)
+	'DebugStop
+	Local first:SToken = params.GetToken(0)
+	Local result:String
+	
+	For Local n:Int = 1 Until params.added
+		result :+ params.getToken(n).value
+	Next
+		
+	Return New SToken( TK_TEXT, result, first.linenum, first.linepos )
+End Function
+
+
+Function SEFN_UCFirst:SToken(params:STokenGroup Var, context:SScriptExpressionContext var)
+	Local value:String = params.GetToken(1).GetValueText()
+	Return New SToken( TK_Text, StringHelper.UCFirst(value), params.GetToken(0) )
+End Function
+
+
+Function SEFN_Csv:SToken(params:STokenGroup Var, context:SScriptExpressionContext var)
+	If params.added <= 1 
+		Return New SToken( TK_ERROR, "No CSV-Entries passed", params.GetToken(0) )
+	EndIf
+
+	Local values:TStringBuilder
+	Local index:Int
+	Local separator:String = ";"
+	Local doTrim:Int = True
+
+	If params.added >= 5
+		local token:SToken = params.GetToken(4)
+		doTrim = TScriptExpression._IsTrueValue( token )
+	EndIf
+	If params.added >= 4
+		separator = String(params.GetToken(3).GetValueText())
+	EndIf
+	If params.added >= 3
+		index = Int(params.GetToken(2).GetValueText())
+	EndIf
+
+	If doTrim
+		'SB's trim() avoids new string creation, so prefer it over string.trim()
+		values = New TStringBuilder(params.GetToken(1).GetValueText()).Trim()
+	Else
+		values = New TStringBuilder(params.GetToken(1).GetValueText())
+	EndIf
+
+	Local splitBuffer:TSplitBuffer = values.Split(separator)
+	If splitBuffer.Length() <= index
+		Return New SToken( TK_ERROR, "CSV-Index too big (" + index + " passed, " + (splitBuffer.Length()+1) + " allowed)", params.GetToken(0) )
+	Else
+		Return New SToken( TK_Text, splitBuffer.Text(index), params.GetToken(0) )
+	EndIf
+End Function
+
+
+' Register the functions
+' The two numbers are MInimum and Maximum number of allowed parameters
+TScriptExpression.RegisterFunctionHandler( "not", SEFN_Not, 1, -1)
+TScriptExpression.RegisterFunctionHandler( "and", SEFN_And, 1, -1)
+TScriptExpression.RegisterFunctionHandler( "or",  SEFN_Or,  1, -1)
+TScriptExpression.RegisterFunctionHandler( "if",  SEFN_If,  1,  3)
+TScriptExpression.RegisterFunctionHandler( "neq",  SEFN_NEq,  1, -1)
+TScriptExpression.RegisterFunctionHandler( "eq",  SEFN_Eq,  1, -1)
+TScriptExpression.RegisterFunctionHandler( "gt",  SEFN_Gt,  2,  2)
+TScriptExpression.RegisterFunctionHandler( "gte", SEFN_Gte, 2,  2)
+TScriptExpression.RegisterFunctionHandler( "lt",  SEFN_Lt,  2,  2)
+TScriptExpression.RegisterFunctionHandler( "lte", SEFN_Lte, 2,  2)
+TScriptExpression.RegisterFunctionHandler( "concat", SEFN_Concat, 2,  2)
+TScriptExpression.RegisterFunctionHandler( "ucfirst", SEFN_UCFirst, 1,  1)
+TScriptExpression.RegisterFunctionHandler( "csv", SEFN_Csv, 2,  3)
```

Comment:
Ich hätte erwartet, dass der zusätzlichen Parameter hier "angemeldet werden muss" also 4 statt 3. Es scheint aber auch so zu gehen...

Reply:
Das wird derzeit eh nicht ausgewertet. Man koennte aber in der TSEFN_Handler.Run()-Methode allgemein die Parametermengen "auswerten" und zentral Fehler auswerfen - statt dies in den individuellen Handlern zu erledigen.

Oder man schmeisst min/max-Param einfach raus...

Reply:
Ich fände es eigentlich gut, wenn die Parameter drin bleiben und zentral ausgewertet würden.
Diese Registrierungsaufrufe erfüllen nämlich auch eine Dokumentationsfunktion. Welche Funktionen gibt es. Die Parameterzahl ist da auch interessant (insb. wenn sie verlässlich ist).